libsolutil/hyperlink.h

```diff
@@ -0,0 +1,72 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <boost/filesystem.hpp>
+
+#include <iostream>
+#include <string>
+
+// Required for gethostname()
+#if defined(_WIN32)
+#include <Winsock2.h>
+#else
+#include <unistd.h>
+#endif
+
+namespace solidity::util
+{
+
+struct Hyperlink
+{
+	bool enabled = false;
+
+	struct Ref { bool enabled; std::string text; };
+
+	Ref operator()(std::string const& _ref) { return Ref{enabled, _ref}; }
+};
+
+inline std::ostream& operator<<(std::ostream& _os, Hyperlink::Ref const& _hyperlink)
+{
+	auto const path = boost::filesystem::path{_hyperlink.text};
+	bool const candidate = boost::filesystem::is_regular_file(path) && (&_os == &std::cout || &_os == &std::cerr);
```

Comment:
I think this heuristic is a bit too crude. We do not retain the actual absolute path to the file throughout the system. You are assuming here that the solidity-internal name of a source file is a relative path starting from the current directory, but the actual logic is much more complicated.

Reply:
It seemed to have worked when compiling uniswap. I was wondering myself too. I could go into the code paths that do fill this sourceName when I'm done with the main task. I just thought it might be helpful functionality for more than just me, so I created a polished PR for it.